aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -36,4 +92,47 @@ Tensor kl_div_backward_cuda(const Tensor& grad, const Tensor& input, const Tenso
   }
   return grad_input;
 }
+
+Tensor binary_cross_entropy_cuda(const Tensor& input, const Tensor& target, const Tensor& weight, int64_t reduction) {
+    Tensor loss;
+    return at::native::binary_cross_entropy_out_cuda(loss, input, target, weight, reduction);
+}
+
+Tensor& binary_cross_entropy_out_cuda(Tensor& loss, const Tensor& input, const Tensor& target, const Tensor& weight, int64_t reduction) {
+  loss = at::empty_like(input);
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(input.scalar_type(), "binary_cross_entropy_out_cuda", [&]() {
+    binary_cross_entropy_out_kernel<scalar_t>(loss, input, target);
+  });
+  if (weight.defined()) {
+    loss.mul_(weight);
+  }
+  if (reduction == at::Reduction::Mean) {
+    loss = loss.mean();
```

Comment:
@ezyang , do I need to fix the way I'm calculating these reductions too? Maybe something like loss.copy_(loss.mean())?

Reply:
Yes, `copy_` should work. If there's a `mean_out` variant you could use that too.